tools/iceoryx_build_test.sh

```diff
@@ -95,6 +97,11 @@ while (( "$#" )); do
         TEST_FLAG="ON"
         shift 1
         ;;
+    "test-add-user")
+        TEST_FLAG="ON"
```

Comment:
I wouldn't enable the test flag here. If you would like to have a test user but no tests it is totally fine. I am just afraid that we go back to our old cmake state where nearly every flag had side effects and some of them enabled stuff and other disabled them.